sw/host/opentitanlib/src/app/mod.rs

```diff
@@ -272,3 +321,44 @@ impl TransportWrapper {
         Ok(())
     }
 }
+
+/// Given an pin/uart/spi/i2c port name, if the name is a known
+/// alias, return the underlying name/number, otherwise return the
+/// string as is.
+fn map_name(map: &HashMap<String, String>, name: &str) -> String {
+    let name = name.to_uppercase();
+    match map.get(&name) {
+        Some(v) => {
+            if v.eq(&name) {
+                name
+            } else {
+                map_name(map, v)
+            }
+        }
+        None => name,
+    }
+}
```

Comment:
This looks like the same function as above? It would have been easier to review if moving functions around was in a separate commit.

Reply:
Yes, my problem, however, is that I am separating a struct into two structs, with some member function going into one and some into the other, and a single one `map_name` needing to be accessed from both, which is why it now sits outside both.  It is quite hard to separate the textual moving around of functions, from the semantic breaking of the struct into two.